packages/main/src/Dialog.ts

```diff
@@ -339,8 +336,16 @@ class Dialog extends Popup {
 		return ICON_PER_STATE[this.state as ValueStateWithIcon];
 	}
 
-	get _role() {
-		return (this.state === ValueState.Error || this.state === ValueState.Warning) ? "alertdialog" : "dialog";
+	get _role(): string | undefined {
+		if (this.accessibleRole === PopupAccessibleRole.None) {
+			return undefined;
+		}
+
+		if (this.state === ValueState.Error || this.state === ValueState.Warning) {
```

Comment:
we should document this behavior - if a user sets a dialog role explicitly, we overwrite it. Is this the desired behavior? If yes, please add a note in the documentation of the property.